File: src/main/java/red/tetracube/smarthub/properties/BrokerConfig.java
package red.tetracube.smarthub.properties;

import org.eclipse.microprofile.config.inject.ConfigProperty;

public interface BrokerConfig {

    @ConfigProperty(name = "client-id")
    String clientId();

    @ConfigProperty(name = "host")
    String host();

    @ConfigProperty(name = "port")
    int port();

    @ConfigProperty(name = "username")
    String username();

    @ConfigProperty(name = "password")
    String password();
}

File: src/main/kotlin/red/tetracube/smarthub/properties/BrokerConfig.kt
package red.tetracube.smarthub.properties;

import org.eclipse.microprofile.config.inject.ConfigProperty

interface BrokerConfig {

    @ConfigProperty(name = "client-id")
    fun clientId(): String

    @ConfigProperty(name = "host")
    fun host(): String

    @ConfigProperty(name = "port")
    fun port(): Int

    @ConfigProperty(name = "username")
    fun username(): String

    @ConfigProperty(name = "password")
    fun password(): String

}
